File: schrottverwaltung/src/main/java/de/fhdortmund/schrottverwaltung/security/AuthController.java
package de.fhdortmund.schrottverwaltung.security;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/auth")
public class AuthController {
    @GetMapping("/validate")
    private void validate(){
        // used to validate if the user is authenticated
    }
}

File: schrottverwaltung/src/main/kotlin/de/fhdortmund/schrottverwaltung/security/AuthController.kt
package de.fhdortmund.schrottverwaltung.security

import de.fhdortmund.schrottverwaltung.user.UserDto
import de.fhdortmund.schrottverwaltung.user.User
import de.fhdortmund.schrottverwaltung.user.UserService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController(

    private val userService: UserService,
) {
    @PostMapping("/login")
    fun authenticateUser(@RequestBody user: UserDto): String = userService.authenticateUser(user)

    @PostMapping("/signup")
    fun registerUser(@RequestBody user: UserDto): User? = userService.registerUser(user)
}